src/oracc_cmd.ts

```diff
@@ -0,0 +1,50 @@
+import * as child_process from 'child_process';
+import { homedir } from 'os';
+import * as path from 'path';
+import * as process from 'process';
+import { promisify } from 'util';
+import * as vscode from 'vscode';
+
+const execFile = promisify(child_process.execFile);
+
+// Thin wrapper around `execFile` which prepares the environment where we want to be running
+// the commands in.
+function run_cmd(file: string,
+                 args: string[] = [],
+                 options: Object = {
+                     cwd: homedir(),
+                     env: {
+                         ...process.env,
+                         ...{PATH: '/usr/local/oracc/bin:' + process.env.PATH},
+                     },
+                 }) {
+    return execFile(file, args, options);
+}
+
+// Run the `oracc` command with the given action.  This assumes the command is being run on
+// the server.
+export async function run_oracc(cmd: string[]) {
+    const editor = vscode.window.activeTextEditor;
+    const ext = path.parse(editor.document.uri.fsPath).ext;
+    if (ext !== ".glo") {
+        vscode.window.showWarningMessage('Oracc command can only be run from glossary files');
+        return;
+    }
+
+    try {
+        const { stdout } = await run_cmd('oracc', cmd);
+        vscode.window.showWarningMessage(`${stdout}`);
+    } catch (err) {
```

Comment:
Could also move this error handling to `view.ts` if you think it's more natural there (where the validation/lemmatisation responses are handled too - although the logic there is a bit more complex).